persist-cli/src/main/java/io/ballerina/persist/utils/ReadBalFiles.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright (c) 2022, WSO2 LLC. (https://www.wso2.com) All Rights Reserved.
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package io.ballerina.persist.utils;
+
+import io.ballerina.compiler.syntax.tree.ModuleMemberDeclarationNode;
+import io.ballerina.persist.nodegenerator.BalSyntaxTreeGenerator;
+import io.ballerina.persist.objects.BalException;
+import io.ballerina.persist.objects.Entity;
+import io.ballerina.persist.objects.EntityMetaData;
+import io.ballerina.projects.DiagnosticResult;
+import io.ballerina.tools.diagnostics.Diagnostic;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static io.ballerina.persist.nodegenerator.BalFileConstants.EXTENSION_BAL;
+import static io.ballerina.persist.nodegenerator.BalFileConstants.KEYWORD_CLIENTS;
+import static io.ballerina.persist.nodegenerator.BalFileConstants.KEYWORD_MODULES;
+import static io.ballerina.persist.nodegenerator.BalFileConstants.PATH_ENTITIES_FILE;
+import static io.ballerina.persist.nodegenerator.BalSyntaxTreeGenerator.formatModuleMembers;
+import static io.ballerina.persist.nodegenerator.BalSyntaxTreeGenerator.generateRelations;
+import static io.ballerina.persist.utils.BalProjectUtils.hasSemanticDiagnostics;
+import static io.ballerina.persist.utils.BalProjectUtils.hasSyntacticDiagnostics;
+
+/**
+ * Class to implement to read the ballerina files.
+ *
+ * @since 0.1.0
+ */
+public class ReadBalFiles {
+
+    public static EntityMetaData readBalFiles(String sourcePath) throws BalException {
```

Comment:
Shall we move this utils function to the `BalProjectUtils` class?
IMO we don't need new utils class